shared/test/test_net_sim_api.rb

```diff
@@ -673,11 +673,44 @@ def test_having_max_routers_on_one_shard_does_not_limit_another
 
     # We should still be able to add a router on another shard
     @shard_id = '_testShard3'
-    create_router_node
+    create_router_node('routerNumber' => CDO.netsim_max_routers + 1)
+    assert_equal(201, @net_sim_api.last_response.status,
+                 'Should have allowed router on another shard')
+    assert_equal(1, read_records(TABLE_NAMES[:node]).count,
+                 'Should have allowed router on another shard')
+  end
+
+  def test_reject_routers_without_router_number
+    create_router_node({})
+    assert_equal(400, @net_sim_api.last_response.status, "Allowed malformed router row")
+  end
+
+  def test_reject_routers_causing_router_number_collision
+    create_router_node('routerNumber' => 1)
     assert_equal(201, @net_sim_api.last_response.status,
-                 "Should have allowed router on another shard")
+                 'Failed to insert first router.')
+
+    # Should return 400 BAD REQUEST when a routerNumber collision is detected
+    # TODO: Should we use 409 CONFLICT instead?
```

Comment:
:disappointed: Started in on this and realized it is complicated by multi-insert.  What if you try to add three rows `[valid, malformed, conflict]` - we reject the entire request, but should we return `400 BAD REQUEST` or `409 CONFLICT` in this case?  Or maybe a 400 with per-row error details?